src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,118 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+
+    private String filePath = ""; // Default is "data/duke.txt"
+
+    /**
+     * Create a storage object.
+     *
+     * @param filePath Specify the location of the file to be saved and loaded from.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Save the current list to a specified file path on the hard disk.
+     *
+     * @param tasks List of tasks to be save.
+     */
+    public void save(ArrayList<Task> tasks) {
+        try {
+            String path = this.filePath;
+            FileWriter fw = new FileWriter(path);
+
+            String text = "";
+            for (int i = 0; i < tasks.size(); i++) {
+                //Get tasks
+                Task task = tasks.get(i);
+
+                if (task instanceof Todo) {
```

Comment:
It may be worth refactoring the logic for each task into an abstract method of Task, implement a concrete implementation for each Class that extends Task, and use polymorphism to get the serialized text, instead of using if/else statements, so that if there is a requirement for different types of tasks in the future, you don't need to touch the code here.